ci/tests/general/file_reader.py

```diff
@@ -0,0 +1,448 @@
+                if debug:
+                    print("Start: '{0}' End: '{1}'".format(search_start, search_end))
+                # Done reading from file
+                self.opened_file.close()
+            # Do not display EOFErrors:
+            except EOFError:
+                pass
+
+        with open(self.FILE_PATH + '_debug', 'w') as debug_file:
+            debug_file.write(str('Will read output'))
+            debug_file.write(str("--- Searching took {0} seconds ---".format(time.time() - start_time)))
+        # Return the contents of the file
+        with open(self.FILE_PATH, 'r') as f:
+            # Print execution time:
+            if debug:
+                print("--- Searching took {0} seconds ---".format(time.time() - start_time))
+            if not suppress_return:
+                # Check if first line isn't empty
+                f.seek(0)
+                if f.readline() == '':
+                    return None
+                return f.read();
+
+    def search_for_errors(self, start, end, python_error=False, paths_to_file=PATHS_TO_FILES, debug=False, host=None,
+                          suppress_return=False):
+        """
+        :param start: Starting date
+        :type start: str / Datetime
+        :param end: End date
+        :type end: str / Datetime
+        :param python_error: Whether only python errors should be checked
+        :type python_error: Boolean
+        :param paths_to_file: List of paths of files that will be searched on all nodes
+        :type paths_to_file: List of str
+        :param debug: Boolean to debug or not
+        :type debug: Boolean
+        :param host: ip of the node
+        :type host: str
+        :param suppress_return: only write to file and not return contents
+        :type suppress_return: Boolean
+        :return: Output of a file as string
+        """
+        if not start:
+            start = (datetime.today() - timedelta(hours=1)).strftime("%Y-%m-%d %H:%M:%S")
+        if not end:
+            end = datetime.today().strftime("%Y-%m-%d %H:%M:%S")
+        python_errors = ['Exception', 'StopIteration', 'StandardError', 'BufferError', 'ArithmeticError',
+                         'FloatingPointError', 'OverflowError', 'ZeroDivisionError', 'AssertionError', 'AttributeError',
+                         'EnvironmentError', 'IOError', 'OSError', 'WindowsError', 'VMSError', 'EOFError',
+                         'ImportError', 'LookupError', 'IndexError', 'KeyError', 'MemoryError', 'NameError',
+                         'UnboundLocalError', 'ReferenceError', 'RuntimeError', 'NotImplementedError', 'SyntaxError',
+                         'IndentationError', 'TabError', 'SystemError', 'TypeError', 'ValueError', 'UnicodeError',
+                         'UnicodeDecodeError', 'UnicodeEncodeError', 'UnicodeTranslateError']
+        errors_patterns = ['not found', 'error', 'something went wrong', ' has no attribute',
```

Comment:
this can indeed be extended with useful info in alba/voldrv logfiles ... e.g. choking / timeouts ...